src/Cake.Core/Tooling/ToolSettings.cs

```diff
@@ -15,6 +15,11 @@ public class ToolSettings
         public FilePath ToolPath { get; set; }
 
         /// <summary>
+        /// Gets or sets optional timeout for tool execution
+        /// </summary>
+        public int? ToolTimeout { get; set; }
```

Comment:
@patriksvensson said...
Also, can we rename to only Timeout?

Should that not be something like `TimeoutInSeconds` or `TimeoutInMilliseconds`?